### ISSUE TITLE
Document provision

### ISSUE THREAD

**Issue by cuducos** [okfn-brasil/jarbas#191]
As #185 got a _LGTM_ but no merge I felt like this project needs to document how provisioning and deploy are currently handled. This might help in code reviewing stuff like that PR but mainly foster provision automation (something we completely lack, although deploy is basically managed by a [Git `post-receive` hook](https://github.com/datasciencebr/jarbas/blob/master/contrib/post-receive)).

I really would love to dismiss the Git hook and this provisioning task but in spite of the efforts (thanks @pedrommone @gomex and @lipemorais) we still miss a Docker architecture for production. I've tested [Dokku](http://dokku.viewdocs.io/dokku/) and it worked pretty well. In a near future this whole provision complexity might be thrown away if @Irio and I can coordinate a schedule to play around with the DNS settings and put Dokku into production.

As neither Docker nor Dokku have happened yet, this is what I remember we need for provisioning a new Jarbas server:

1. SSH accessible server, let's say, `myjarbas`
1. A non-root user, let's say, `jarbas` user
1. Python 3.5, Postgres, nginx, Yarn and memcached installed
1. A git repository at `/opt/jarbas/` our project directory
1. A bare git repository (not `git init`, `git init --bare`) at `/opt/jarbas.git/`
1. A `virtualenv` at `/opt/jarbas.venv/`
1. A database and user at our Postgres
1. Create a set of envvars at  `/opt/jarbas/.env` following `contrib/.env.sample` and the [settings instructions](https://github.com/datasciencebr/jarbas#settings)
1. Copy `contrib/post-receive` to `/opt/jarbas.git/hooks/post-receive`
1. Give execution permission to `/opt/jarbas.git/hooks/post-receive` (that is to say `chown a+x`)
1. Copy `contrib/nginx` to `/etc/nginx/sites-enabled` (with a more meaningful name) and restart `nginx`

Now what we do in our development machine:

```console
$ git remote add production jarbas@myjarbas:/opt/jarbas.git
$ git push production master
```

Yay… finally you've made your first Jarbas deploy.

---

Probably I forgot some steps, but LMK if you try it so I can fix updating this post.

**Comment by anaschwendler** [comment]
So after a conversation we decided by creating a droplet that will be made for running as a test.
Soon I and @jtemporal will be working on it! ;)

**Comment by cuducos** [comment]
In addition currently there are three tasks added to `crontab`:

```crontab
0 0 * * 0-6 /opt/jarbas.venv/bin/python /opt/jarbas/manage.py receipts --pause 30
0 0 * * 0-6 /opt/jarbas.venv/bin/python /opt/jarbas/manage.py tweets
0 0 1 * * certbot renew
```

**Comment by gomex** [comment]
Can we schedule some hack{day|night} to work pairing on automate that (using Docker or not)?

**Comment by jtemporal** [comment]
>Can we schedule some hack{day|night} to work pairing on automate that

PLEASE!

**Comment by lipemorais** [comment]
I would love to colaborate to Jarbas been deployed smoothly. @gomex @jtemporal @cuducos

**Comment by cuducos** [comment]
I love this idea. @gomex and @lipemorais I _guess_ you might have a more tight schedule than @jtemporal and I. Can you suggest three days and times so we can check our schedules and pick up one?

**Comment by gomex** [comment]
@cuducos and @jtemporal  can you please add your possible days and time here: http://doodle.com/pt_BR/ we can choose together! :)

**Comment by lipemorais** [comment]
This is a suggestion. What do you think? 
https://doodle.com/poll/iix8n7wxshd3yfac

**Comment by lipemorais** [comment]
Looks that we have a winner date and some fall backs. :)
@gomex @jtemporal @cuducos 

![image](https://user-images.githubusercontent.com/5021565/28834036-91f298e4-76b7-11e7-9858-9726145ddc7b.png)

**Comment by cuducos** [comment]
Hell yeah! Scheduled it is — all in?

**Comment by gomex** [comment]
YES! See you tomorrow!

**Comment by cuducos** [comment]
Closed by #293
